R/exploratoryfactoranalysis.R

```diff
@@ -892,17 +892,54 @@ exploratoryFactorAnalysisInternal <- function(jaspResults, dataset, options, ...
 
 }
 
-.efaAddComponentsToData <- function(jaspResults, modelContainer, options, ready) {
-  if(!ready || !options[["addPC"]] || options[["PCPrefix"]] == "" || modelContainer$getError()) return()
+.efaAddFactorsToData <- function(jaspResults, modelContainer, options, ready) {
+
+  if (!ready ||
+      !is.null(jaspResults[["addedScoresContainer"]]) ||
+      modelContainer$getError() ||
+      !options[["addFactorScores"]])
+  {
+    return()
+  }
+
+  colNamesR <- gettextf("Factor_%s", seq_len(length(options$variables)))
+
+  container    <- createJaspContainer()
+  container$dependOn(optionsFromObject = modelContainer, options = "addFactorScores")
 
   scores <- modelContainer[["model"]][["object"]][["scores"]]
 
-  for (i in 1:ncol(scores)) {
-    scorename <- paste0(options[["PCPrefix"]], "_", i)
-    if (is.null(jaspResults[[scorename]])) {
-      jaspResults[[scorename]] <- createJaspColumn(scorename)
-      jaspResults[[scorename]]$dependOn(optionsFromObject = modelContainer)
-      jaspResults[[scorename]]$setScale(scores[, i])
+  for (ii in seq_len(ncol(scores))) {
+
+    colNameR <- colNamesR[ii]
+
+    if (jaspBase:::columnExists(colNameR) && !jaspBase:::columnIsMine(colNameR)) {
+      .quitAnalysis(gettext("Column name already exists in the dataset"))
```

Comment:
I think it would make sense to also say which variable name already exists (otherwise it might be a bit confusing for users with very wide datasets)